src/time/Waveform.cpp

```diff
@@ -75,5 +85,29 @@ Eigen::VectorXd Waveform::extrapolateData(int order, int timeWindows)
   return extrapolatedValue;
 }
 
+Eigen::VectorXd Waveform::interpolateData(int order, int timeWindows, double dt)
```

Comment:
you use the name "timeWindows" for two things here -> confusing

Reply:
This part of the interface I never really liked. I will try to get rid of this parameter. Idea: The waveform should track on it's own how many valid samples it has. Might also be worth putting this into a small independent PR, since it will already affect and simplify `Waveform::extrapolateData` and how it is used by the `BaseCouplingScheme`.

Reply:
Number of valid samples is now tracked inside of `Waveform`. See 71e2753. I'll create a PR implementing these changes directly on develop in a moment.

Reply:
Done in https://github.com/precice/precice/pull/1083.